src/server/batch-controller.js

```diff
@@ -51,10 +74,23 @@ class BatchController {
   registerApi(router) {
     router.get(
       '/api/v1/batches/:batchId',
-      needsPermission(permissions.VIEW_BATCHES),
+      needsPermission(permissions.MANAGE_BATCHES),
```

Comment:
Very nice split of the admin controller into the actually responsible controllers with dedicated clearly named permission!
Makes things much more clear.

Reply:
Yeah, we did not think through permissions a lot, so we went very mixed, this consolidates it a little bit more.